cleo/application.py

```diff
@@ -325,6 +325,12 @@ def run(
 
             try:
                 exit_code = self._run(io)
+            except BrokenPipeError:
+                # If we are piped to another process, it may close early and send a
+                # SIGPIPE: https://docs.python.org/3/library/signal.html#note-on-sigpipe
+                devnull = os.open(os.devnull, os.O_WRONLY)
+                os.dup2(devnull, sys.stdout.fileno())
+                exit_code = 1
```

Comment:
why exit with error?

I would expect that the result of `poetry -V | head -1 ; echo $?` would be zero.

Reply:
agreed, updated